### ISSUE TITLE
WARNING - Shift amount out of bounds

### ISSUE THREAD

**Issue by vobruba-martin** [google/closure-compiler#2101]
``` javascript
/**
 * @param {number} n
 * @return {number}
 */
function pow2(n) {
    if (n >= 0 && n < 31) {
        return 1 << n;
    }
    return Math.pow(2, n);
}

console.log(pow2(32));
```

```
media/js/test.js:7: WARNING - Shift amount out of bounds (see right operand): LSH 7 [length: 6] [source_file: media/js/test.js] : number
        return 1 << n;
               ^
```

I wish Closure Compiler was smarter in this example.

**Comment by vobruba-martin** [comment]
BTW I tried to suppress the warning via `@suppress {transitionalSuspiciousCodeWarnings}` but it is not working :(

**Comment by dimvar** [comment]
We recently had a similar discussion about the bitwise operators. See issue #1852. IMO, we should just get rid of this warning as well.

**Comment by vobruba-martin** [comment]
+1 Would be great if this one is removed too. Thanks!
